src/dune_2_excel.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+from typing import List, Any
+from datetime import datetime
+
+import click
+from dotenv import load_dotenv
+from dune_client.client import DuneClient
+from dune_client.models import ResultsResponse
+from dune_client.query import Query
+from dune_client.types import QueryParameter
+import pandas as pd
+from tqdm import tqdm
+
+
+@click.command()
+@click.option("--query", "-q", multiple=True)
+@click.option("--start-date", "-s", type=click.DateTime(formats=["%Y-%m-%d"]))
+def main(query: List[str], start_date: datetime) -> Any:
+    """
+    Main function of the script
+    Args:
+        query: List of queries to fetch
+        start_date: start_date query parameter
+
+    Returns:
+
+    """
+    monthly_reporting(query, start_date)
+    return 0
+
+
+def store_results(results: List[ResultsResponse], start_date: datetime) -> None:
+    """
+    Store results into xlsx file
+    Args:
+        results: results to be stored
+        start_date: start_date query parameter
+    """
+    writer = pd.ExcelWriter(  # pylint: disable=abstract-class-instantiated
+        f'out/{"_".join([str(x.query_id) for x in results])}'
+        f'_{start_date.strftime("%Y-%m-%d")}.xlsx',
+        engine="xlsxwriter",
+    )
+    for result in results:
+        pd.DataFrame(pd.DataFrame(result.result.rows)).to_excel(
+            writer, sheet_name=f"{result.query_id}", index=False
+        )
+    writer.close()
+
+
+def fetch_results(queries: List[str], start_date: datetime) -> List[ResultsResponse]:
+    """
+    Fetches results from Dune
+    Args:
+        queries: List of queries to fetch
+        start_date: start_date query parameter
+
+    Returns:
+        List[ResultsResponse]
+    """
+    dune_client: DuneClient = DuneClient(os.environ["DUNE_API_KEY"])
+    results: List[ResultsResponse] = []
+    for q in tqdm(queries):
+        query = Query(
+            name="",
+            query_id=int(q),
+            params=[QueryParameter.date_type("StartTime", start_date)],
+        )
+        result = dune_client.refresh(query)
+        results.append(result.get_rows())
```

Comment:
src/dune_2_excel.py:72: error: Argument 1 to "append" of "list" has incompatible type "List[Dict[str, str]]"; expected "ResultsResponse"

Not sure if append is what you are looking for here... probably want 
```
results += result.get_rows()
```

Reply:
Yeah I'm not actually looking for the rows here just the results response. Rows will be taken out in a later step. Fixed it in a later commit.